src/app/shared/entry.ts

```diff
@@ -196,4 +197,55 @@ export abstract class Entry implements OnInit, OnDestroy, AfterViewInit {
     }
     return selectedVersion;
   }
+
+  // Selects a tab of index tabIndex
+  selectTab(tabIndex: number) {
+    this.entryTabs.tabs[tabIndex].active = true;
+  }
+
+  // Sorts two entries by last modified, and then verified
+  entryVersionSorting(a: any, b: any): number {
+    if (a.last_modified > b.last_modified) {
+      return -1;
+    } else if (a.last_modified < b.last_modified) {
+      return 1;
+    } else {
+      if (a.verified && !b.verified) {
+        return -1;
+      } else if (!a.verified && b.verified) {
+        return 1;
+      } else {
+        return 0;
+      }
+    }
+  }
+
+  // Sorts a list of versions by last modified date and verified, returning a subset of the versions
+  // If a default version is set, it will appear at the top
+  // If not, will set to latest
+  getSortedVersions(versions: Array<any>, defaultVersion: any): Array<any> {
```

Comment:
This seems to overuse `any` when these appear to always be tags.

Reply:
Technically it is tags or workflowversion, I wasn't sure if there was a generic version type.

Reply:
Type unions might the be solution.  `defaultVersions: Tag | WorkflowVersion`

Reply:
Yeah, the lack of inheritance in the generated swagger classes is annoying that way, but the type unions sound interesting